src/server/helpers/middleware.ts

```diff
@@ -72,6 +74,37 @@ export const loadLanguages = makeLoader('Language', 'languages', (a, b) => {
 	return a.name.localeCompare(b.name);
 });
 
+export function loadEntities(req: $Request, res: $Response, next: NextFunction) {
```

Comment:
The function name is a bit confusing; it promises loading entities but then the code is all about relationships and sorting attributes.

I'm not entirely sure from reading the method why it is needed. Maybe you could explain the steps you took?

Reply:
Yes indeed it's very confusing. I have renamed it now to `loadSeriesItems` .
- This function first extracts the deeply nested relationship **number** and **position** attribute value and attaches directly to the relationship object. so that we can easily access it (e.g: relationship.number or relationship.position) https://github.com/bookbrainz/bookbrainz-site/pull/649/files#diff-8f13ac13a037cddb521ab9b4829f038c0021e5e761d544b826cb539b70682c77R84
- Secondly we perform sort operations on either number or position attribute of the relationship object depending on the ordering type. https://github.com/bookbrainz/bookbrainz-site/pull/649/files#diff-8f13ac13a037cddb521ab9b4829f038c0021e5e761d544b826cb539b70682c77R84
- At last we extract the required properties from the relationship object to an array **seriesItems** used to display the series items in the entity-table. https://github.com/bookbrainz/bookbrainz-site/pull/649/files#diff-8f13ac13a037cddb521ab9b4829f038c0021e5e761d544b826cb539b70682c77R94-R99